### ISSUE TITLE
Error on update Foreign Key (AccountId)

### ISSUE THREAD

**Issue by torehc** [django-salesforce/django-salesforce#283]
Hi, when trying to update the FK I get the following error:
```
Contact.objects.filter(...).update(account=Account.objects.get(...))
```
`salesforce.dbapi.exceptions.SalesforceError: INVALID_FIELD_FOR_INSERT_UPDATE
    Account: bad field names on insert/update call:  AccountId
    FIELDS: [' AccountId']`
    
I'm trying to change the "account" field of the Contact to another, but without success.
``` python
class Contact(models.Model):
    ...
    account = models.ForeignKey('Account', models.DO_NOTHING, blank=True, null=True)
    ...
```
The models have been obtained with inspectdb.

**Comment by hynekcer** [comment]
I can not reproduce this bug with the current django-salesforce v3.2 neither with the newest Django 3.2 or with the oldest supported Django 2.0.

Please specify the versions of both packages. Write also the type of conditions in the filter and verify that the Account field of Contact can be updated by a normal `.save()` method. Maybe your organization have a validation rule installed on Contact that is triggered on Contact update. (The condition in `Account.objects.get(...)` is not important because it is resolved completely before `update(...)`, but maybe some Account can't be assigned due to your business logic rules.)

---
I found a different bug in a case `Contact.objects.filter(account=None).update(account=some_account)`.
It can be easily bypassed by `Contact.objects.filter(account_id='').update(account=some_account)`.
However I will try to fix that special case, even that it could be found finally as an edge case bug in Django.

**Comment by hynekcer** [comment]
The error "SalesforceError: INVALID_FIELD_FOR_INSERT_UPDATE" can be reproduced if the logged user does not have the "Edit Access" permission for the "Account Name" field of "Contact" object. That is self-evident.

**Comment by hynekcer** [comment]
The original issue seems invalid now. The secondary problem is fixed by PR #284.

**Comment by torehc** [comment]
You're right, it was a permissions problem in the organization. Thank you very much and sorry for the inconvenience.
